eos-split-flatpak-repo.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Split Flatpak repo from system OSTree repo
+DefaultDependencies=no
+Conflicts=shutdown.target
+Wants=local-fs.target
+After=local-fs.target
+
+# This is only needed when the flatpak repo is a symlink
+ConditionPathIsSymbolicLink=/var/lib/flatpak/repo
+
+# Try to run before any ostree and flatpak clients so that the repo can
+# be exclusively locked.
+Before=eos-autoupdater.service eos-updater.service
+Before=eos-update-flatpak-repos.service
+Before=eos-updater-avahi.service eos-update-server.service
+Before=eos-updater-flatpak-installer.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+ExecStart=/usr/sbin/eos-split-flatpak-repos
```

Comment:
If `--prune` is not passed by default, when does pruning happen?

Reply:
Good question, this should probably be discussed somewhere here. The repos would be pruned the next time something else prunes the repos. So long as the refs are removed, the objects will be cleaned up. This would happen when eos-updater deploys a new OS and gnome-software/flatpak update or install a flatpak in the system repo. The presumption is that would happen sooner or later, but obviously there's no guarantee.

The idea is to make this operation fast since the prune would be the slow part. Since the objects are all hard linked during the split, it won't initially use any extra disk space. However, until both repos have been pruned there will be some excess. For instance, if you deployed a new OS and `/ostree/repo` was pruned, it would cleanup both the unused flatpak objects as well as the non-rollback OS objects. However, until `/var/lib/flatpak/repo` was pruned, those non-rollback OS objects would still exist on disk. Maybe this is a premature optimization, but I've only run this on smallish repos.

Reply:
As long as flatpak and eos-updater both prune after doing something which makes the repository content significantly change, I think this approach is fine. It would be good to have it documented in a comment somewhere (you could probably copy/paste what you wrote above).